controller/src/beerocks/master/db/db.cpp

```diff
@@ -1232,6 +1232,48 @@ const std::vector<beerocks_message::sWifiChannel> db::get_hostap_supported_chann
     return n->hostap->supported_channels;
 }
 
+/**
+ * @brief Add supported operating class to the database.
+ * Currently this function is a wrapper which converts the operating
+ * class to a set of supported channels and updates the list of currently
+ * supported channels.
+ * 
+ * @param mac radio mac
+ * @param info operating class info from the tlvApRadioBasicCapabilities TLV
+ * @return true on success
+ * @return false on failure
+ */
+bool db::add_hostap_supported_operating_class(std::string mac, wfa_map::cOperatingClassesInfo &info)
+{
+    auto supported_channels = get_hostap_supported_channels(mac);
+    auto channel_set = wireless_utils::operating_class_to_channel_set(info.operating_class());
+    std::unordered_set<uint8_t> affected_channels;
+    auto non_oper_channels_list_length = info.statically_non_operable_channels_list_length();
+    for (int ch_idx = 0; ch_idx < non_oper_channels_list_length; ch_idx++) {
+        auto ch_tuple = info.statically_non_operable_channels_list(ch_idx);
+        auto ch       = std::get<1>(ch_tuple);
+        affected_channels.insert(ch);
+    }
+
+    for (auto c : channel_set) {
+        // First, delete old channel if exists
+        auto old =
+            std::find_if(supported_channels.begin(), supported_channels.end(),
+                         [&](const beerocks_message::sWifiChannel ch) { return ch.channel == c; });
```

Comment:
Might be worth to make channel_set a map instead of a set.

Reply:
I don't understand why - Do you mean it will be simpler if instead of set we use a 1 to 1 mapping - something like a lookup table? so instead of `find_if` we can simply access `channel_map[channel]` to see if it exists?